apis/r/tests/testthat/test-query-condition.R

```diff
@@ -0,0 +1,184 @@
+test_that("DataFrame Factory", {
+    skip_if(!extended_tests())
+    uri <- tempfile()
+    if (dir.exists(uri)) unlink(uri, recursive=TRUE)
+
+    ctx <- soma_context()
+
+    sch <- arrow::schema(
+      arrow::field("soma_joinid", arrow::int64()),
+      arrow::field("int8",   arrow::int8()),
+      arrow::field("int16",  arrow::int16()),
+      arrow::field("int32",  arrow::int32()),
+      arrow::field("int64",  arrow::int64()),
+      arrow::field("uint8",  arrow::uint8()),
+      arrow::field("uint16", arrow::uint16()),
+      arrow::field("uint32", arrow::uint32()),
+      arrow::field("uint64", arrow::uint64()),
+      arrow::field("string", arrow::string()),
+      arrow::field("large_utf8", arrow::large_utf8()),
+      arrow::field("enum",
+          arrow::dictionary(
+              index_type = arrow::int8(),
+              value_type = arrow::utf8(),
+              ordered = TRUE)),
+      arrow::field("float32", arrow::float32()),
+      arrow::field("float64", arrow::float64())
+      # TODO: datetime types
+    )
+
+    sdf <- SOMADataFrameCreate(uri, sch, index_column_names = "soma_joinid")
+    expect_true(sdf$exists())
+    expect_true(dir.exists(uri))
+
+    tbl <- arrow::arrow_table(
+        soma_joinid = 1L:10L,
+        int8   =  -11L:-20L,
+        int16  = -201L:-210L,
+        int32  = -301L:-310L,
+        int64  = -401L:-410L,
+        uint8  =   11L:20L,
+        uint16 =  201L:210L,
+        uint32 =  301L:310L,
+        uint64 =  401L:410L,
+        string = c("apple", "ball", "cat", "dog", "egg", "fig", "goose", "hay", "ice", "jam"),
+        large_utf8 = c("APPLE", "BALL", "CAT", "DOG", "EGG", "FIG", "GOOSE", "HAY", "ICE", "JAM"),
+        enum = factor(
+            c("red", "yellow", "green", "red", "red", "red", "yellow", "green", "red", "green"),
+            levels = c("red", "yellow", "green")),
+        float32 = 1.5:10.5,
+        float64 = 11.5:20.5,
+        schema = sch)
+    sdf$write(tbl)
+    sdf$close()
+
+    sdf$reopen("READ")
+
+    good_cases <- list(
+      'soma_joinid > 5' = function(df) {
+          expect_equal(df$soma_joinid, 6:10)
+          expect_equal(df$int32, -306:-310)
+      },
+      'soma_joinid == 10' = function(df) {
+          expect_equal(df$soma_joinid, 10)
+          expect_equal(df$int32, -310)
+          # XXX FIXME
+          # 'current' is not a factor
+          # expect_equal(df$enum, c("green"))
```

Comment:
@mojaveazure I'm having a hard time reading back factors and checking them against expected values. More in Slack if you like.

Reply:
Resolved thanks to @mojaveazure: 

```
expect_equal(as.character(df$enum), c("green"))
```